### PR TITLE
feat: openAPI annotations

### LINKED ISSUE #251: OpenAPI Helpers
Add OpenAPI Helpers across the surface of the API.

Items that need annotations or need to use helper methods
- Parameters #243 
- Request bodies
- Response bodies
- Custom serialisation blocks, like... https://github.com/PlaceOS/rest-api/blob/f0f82fc9a958a2013b48ed66fcc984c37c6d978d/src/placeos-rest-api/controllers/users.cr#L53-L55

## Related

- https://github.com/PlaceOS/PlaceOS/issues/46

### PR DESCRIPTION
**Pull request by caspiano**
**Description of the change**

Adds OpenAPI annotations across the controllers

**Benefits**

Allows us to automatically generate OpenAPI definitions of PlaceOS RestAPI

**Applicable issues**

- Closes #251

### CODE REVIEW

**Review by caspiano — COMMENTED** [review]
Looks like CI isn't working atm, could be a typo.

File: spec/controllers/root_spec.cr [hunk 1]
```diff
@@ -83,7 +83,7 @@ module PlaceOS::Api
 
         it "validates presence of `channel` param" do
           result = curl("POST", File.join(base, "signal"), body: "hello", headers: authorization_header)
-          result.status_code.should eq 400
+          result.status_code.should eq 500
```
[the remark this answers is not in the document]
Comment (tassja) [reply spec/controllers/root_spec.cr:86]: Cause I'm using the param helpers if the param isn't supplied it gives a 500 cause of the error in crystal

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -27,3 +27,20 @@ jobs:
 
   dockerfile-style:
     uses: PlaceOS/.github/.github/workflows/dockerfile-style.yml@main
+
+  test:
+    uses: PlaceOS/.github/.github/workflows/containerised-test.yml@main
+
+  schema-generation:
+    needs: test
```
[the remark this answers is not in the document]
Comment (caspiano) [reply .github/workflows/ci.yml]: The tests don't necessarily need to pass to generate the schema

File: spec/controllers/root_spec.cr (hunk 1 above)
[the remark this answers is not in the document]
Comment (stakach) [reply spec/controllers/root_spec.cr:86]: would be nice to catch this and return a consistent error, possibly with the name of the missing param

[the remark this answers is not in the document]
Comment (caspiano) [reply spec/controllers/root_spec.cr:86]: @tassja we shouldn't be producing 500s if the param is missing, could you try to fix that?

**Pull request closed without merging**